### PR TITLE
(#361) Upgrade to Gatsby 5

### LINKED ISSUE #361: Enabler: Update the documentation site to use Gatsby 5
# Title of Enabler
> **ESTIMATE** TBD

Gatsby 5 has been out since the end of 2022. 3.x had been unsupported since 5.x release in 2022. (Not that we have kept up to date with 3.x)

### PR DESCRIPTION
**Pull request by bryanpizzillo**
## Pull Request Details
Closes #361

This PR upgrades Gatsby to version 5. Outstanding action is to fix hot-reloading.

Major Changes:
* The MDX processor change a lot which introduced all kinds of things. Specifically:
   * MDX now process ALL html (except `<a>` because we did magic) as JSX. This means it can't be HTML. This also means raw html does not get processed by the markdown wrapper anymore.  (So an `<h2>` tag will not be handled the same as `##` markdown.
   * You must add the language to code fences (that is 3 backticks)
   * You now must handle `nopreview` for HTML by making the code language `html-nopreview`.
      * Technically everything supports `-nopreview` but only HTML would actually preview.
   * DON'T use "html" unless you have to. That html becomes JSX. This means do not add `<h2>` tags because they will not get the anchor link treatment.
   * Images referenced by MDX should be placed in the content folder. Images referenced in content within the mdx will be processed by gatsby-remark-image.
   * GraphQL Matters now. 
      * When adding new properties to YAML frontmatter, you *must* add then to the schema in gatsby-node.js AND update any queries to specifically reference that new field.
      * There is no longer a single query in gatsby-node.js. createPages now queries only the basic information needed to call createPage. 
      * The queries for the pages are actually in the layout templates. (default-layout.jsx, component-page-layout.jsx, and utility-page-layout.jsx) 
         * Technically default-layout.jsx does not need a query because it has all the data it needs in the MDX.

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [x] Unit tests have been updated or created to reflect any javascript changes.
- [x] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [x] WCAG 2.1 Level AA requirements have been met.

### Development:
- [x] Any new or updated javascript code has 100% unit test coverage.
- [x] New or updated breakpoints have regression images.
- [x] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [x] There are no unexpected or unapproved regression image changes. 
- [x] Functionality of interactive elements meet the acceptance criteria.
- [x] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [x] Lighthouse: scores have not noticeably decreased during this PR.
- [x] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [x] New or updated features introduced in this PR are developed mobile-first.
- [x] Breakpoint changes and regression images match those breakpoints.
- [x] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [x] New code passes code quality standards set by industry standards.
- [x] The expected Storybook stories have been added or updated for the new or updated feature.
- [x] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [x] VoiceOver: Described content matches with what was expected.
- [x] Keyboard navigation: new or updated features and content are navigable via the keyboard.

### CODE REVIEW

**Review by olitharp-nci — APPROVED** [review]

**Comment by KateMashkinaNIH** [comment]
1. console logs and Minified React error
Screenshot 2024-08-13 at 5 03 16 PM
2. Missing components page description ( all components pages) 
Screenshot 2024-08-13 at 5 08 10 PM
3. missing p tag that's wrapping images is causing them to not be centered  ( example /get-started/figma-guide page) 
4. command line code is displayed in black and code doesn't have color 
Screenshot 2024-08-13 at 5 14 23 PM
Screenshot 2024-08-13 at 5 13 32 PM
5. gutters header doesn't have an href ( /foundations/layout-positioning/layout-grid#gutters ) so its not actionable 
Screenshot 2024-08-13 at 5 17 01 PM

**Comment by bryanpizzillo** [comment]
Notes, and primarily for me.

- [ ] 1.  This error is because the hydrated state of the page does not match the rendered by react version. It looks like this only happens on pages with side navs, and I can visually see the page getting rendered by the browser and then re-rendered by react to show the nav. The nav should be part of the rendered HTML.
- [ ] 2. Easiest fix of the bunch, description was just missing from the GraphQL query for component layout.
- [ ] 3. Images are, um, challenging. The images on the figma page are in the content folder, and are getting treated special because of that. Only the homepage is getting handled by Markdown because it is referencing an image from the static folder. The other images are doing something else. Probably the sharp transformer or something. Anyway, these used to get `<p>` now they don't, but they are not handled by markdown. Also they have pretty crappy alt tags, I mean really bad. Like gatsby just took the file name and removed dashes. Anyway, this is probably a CSS thing to style `.gatsby-resp-image-wrapper` which is what is getting rendered. This does leave us wondering if the home page image should go into the mdx folder... 
- [ ] 4. `prism-react-renderer` changed the default languages it ships with to a much shorter list than before. There is some [BS way](https://github.com/FormidableLabs/prism-react-renderer?tab=readme-ov-file#custom-language-support) to add additional languages that will probably take 2 or 3 hours to figure out how to actually get working correctly. 
   * The spacing issues I have not found yet.
- [ ] 5. Gutters not getting the link for doing the anchor thing is unknown. I see others on the layout-grid.mdx page not working either. Responsive Variants and Auto Layout Columns being 2 not found. 

@KateMashkinaNIH you showed me a regression image yesterday that was showing the "Component Preview" heading that should not have. What page was that?

**Comment by KateMashkinaNIH** [comment]
@bryanpizzillo its /foundations/typography/fonts ( updated my comment to note that as well)

**Comment by KateMashkinaNIH** [comment]
Round 2:

  first description is missing from code snippet 
https://designsystem-dev.cancer.gov/pr-1820/components/accordion
https://designsystem-dev.cancer.gov/pr-1820/components/collection
https://designsystem-dev.cancer.gov/pr-1820/components/combo-box
https://designsystem-dev.cancer.gov/pr-1820/components/footer
https://designsystem-dev.cancer.gov/pr-1820/components/header
https://designsystem-dev.cancer.gov/pr-1820/components/header/mega-menu
https://designsystem-dev.cancer.gov/pr-1820/components/site-alert
Screenshot 2024-08-15 at 12 38 44 PM

File: docs/content/foundations/typography/fonts.mdx [hunk 1]
```diff
@@ -115,7 +115,7 @@ To use the following Google fonts in your site, add the follow block into your `
 
 ### Recommended Way to Use Fonts
 
-```html nopreview
+```html-nopreview
```
Comment (olitharp-nci) [inline docs/content/foundations/typography/fonts.mdx:118]: huh?

File: docs/gatsby-config.mjs [hunk 2]
```diff
@@ -54,22 +54,28 @@ const config = {
...
-		{
-			resolve: 'gatsby-plugin-react-svg',
-			options: {
-				rule: {
-					include: '/icons/',
-				},
-			},
-		},
-		'gatsby-transformer-yaml',
+		// {
```
Comment (olitharp-nci) [inline docs/gatsby-config.mjs:57]: wanna remove this?

File: docs/content/foundations/token-changes.mdx [hunk 3]
```diff
@@ -43,14 +43,14 @@ USWDS is built using a suite of [customizable settings](https://designsystem.dig
...
-| `'shadow-4'`    | `'box-shadow: 0 12px 24px 0 rgba(0, 0, 0, 0.1)'`<br/>`'box-shadow: 0 12px 24px 0 rgba(0, 0, 0, 0.3)'` | <div style="box-shadow: 0 1px 24px 0 rgba(0, 0, 0, 0.1)">Old Shadow 4</div><br/><div className="shadow-4">New Shadow 4</div> |
-| `'shadow-5'`    | `'box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.1)'`<br/>`'box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.3)'` | <div style="box-shadow: 0 1px 32px 0 rgba(0, 0, 0, 0.1)">Old Shadow 5</div><br/><div className="shadow-5">New Shadow 5</div> |
+| Shadow Token    | CSS Values (Old to New)                                                                               | Visualization                                                                                          |
+| :-------------- | :---------------------------------------------------------------------------------------------------- | :----------------------------------------------------------------------------------------------------- |
```
Comment (olitharp-nci) [inline docs/content/foundations/token-changes.mdx:47]: weren't  you getting rid of the colons?

**Review by bennettcc — APPROVED** [review]

**Comment by bennettcc** [comment]
Product and QA approve this PR as of 8/16. Sarina, Andy, Kate, Nadia, Monika, and Adriana reviewed and approved

File: docs/gatsby-config.mjs (hunk 2 above)
Comment (bryanpizzillo) [reply docs/gatsby-config.mjs]: They have been removed from both the config and the packages list.

File: docs/content/foundations/token-changes.mdx (hunk 3 above)
Comment (bryanpizzillo) [reply docs/content/foundations/token-changes.mdx:47]: I only changed the one table that broke. This did not seem to break.

**Pull request merged**